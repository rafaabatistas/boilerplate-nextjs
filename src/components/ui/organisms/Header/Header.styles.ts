import styled from 'styled-components';

export const Header = styled.header`
  width: 100vw;
  height: 80px;
  background-color: #19062b;
  display: flex;
  align-items: center;
  justify-content: center;
  border-bottom: 1px solid white;
`;
