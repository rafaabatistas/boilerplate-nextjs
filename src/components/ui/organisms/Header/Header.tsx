import * as S from './Header.styles';

import { NavList } from '~molecules/NavList/NavList';

export const Header = () => {
  return (
    <S.Header>
      <NavList />
    </S.Header>
  );
};
