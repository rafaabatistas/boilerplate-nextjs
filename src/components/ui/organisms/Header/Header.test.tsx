import { render } from '@testing-library/react';

import { Header } from './Header';

describe('<Header />', () => {
  it('Deve renderizar o componente <Header />', () => {
    //Arrange
    const { container } = render(<Header />);
    //Assert
    expect(container.firstChild).toMatchSnapshot();
  });
});
