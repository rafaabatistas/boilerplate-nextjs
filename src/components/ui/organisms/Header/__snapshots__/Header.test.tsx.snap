// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Deve renderizar o componente <Header /> 1`] = `
.c0 {
  width: 100vw;
  height: 80px;
  background-color: #19062b;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-bottom: 1px solid white;
}

.c1 {
  width: 100%;
  height: 100%;
  max-width: 1240px;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 50px;
}

.c2 {
  display: block;
  width: 60px;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  width: 50px;
}

.c4 {
  color: white;
  font-size: 1.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.5s ease-out;
  transition: all 0.5s ease-out;
}

.c4:hover {
  color: #adadad;
}

<header
  class="c0"
>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      <img
        alt="Imagem escrito nextJS"
        class="c3"
        src="assets/img/icon-react.png"
      />
    </li>
    <li
      class="c2"
    >
      <a
        class="c4"
        href="https://github.com/rafaabatistas/boilerplate-nextjs"
        rel="noreferrer noopener"
        target="_blank"
      >
        GitHub
      </a>
    </li>
  </ul>
</header>
`;
